### DISCUSSION TITLE
Problems with figures generated by scv.pl.velocity_graph(adata)

### DISCUSSION THREAD

**Discussion by ETAOIK** [theislab/scvelo#995] [Q&A]
Hi, I am new to scVelo application and I have some problems during studying the tutorial 'Velocity graph and pseudotime' in https://scvelo.readthedocs.io/en/stable/VelocityBasics/, 
In my output, cell-to-cell transitions didn't overlap with the umap clusters plot like following(Fig.1), 
scv.pl.velocity_graph(adata,threshold=0.2)
![emmm](https://user-images.githubusercontent.com/82143598/215980660-07d2893c-dbed-486c-ad6d-e9340406fa08.png)
only set arrows=True could made it like the following(Fig.2), and in this situation edges_on_top=None didn't works
![threshold0](https://user-images.githubusercontent.com/82143598/215981427-1ad68c0f-4742-4675-b869-2d34d85108ea.png)

also, the the velocity graph generated by scv.pl.velocity_graph(adata) is quite small and the legend is quite big(Fig.3)

x, y = scv.utils.get_cell_transitions(adata, basis='umap', starting_cell=70)
ax = scv.pl.velocity_graph(adata, c='lightgrey', edge_width=.05, show=False)
ax = scv.pl.scatter(adata, x=x, y=y, s=120, c='ascending', cmap='gnuplot', ax=ax)
![emmm2](https://user-images.githubusercontent.com/82143598/215982816-656e6573-c68d-4927-b139-390115eb2b4d.png)

My question is how to deal with it. Any help is greatly appreciated!

**Comment by joe-jhou2** [comment]
I run into the same problem. Try to modify some parameter, but nothing changed.

**Reply by ETAOIK** [reply]
I modified the source codes, and found one solution, although it didn't solve the problems fundamentally (cause if you didn't set `arrows=True`, the changes would not work), so here is my solution:
modify the codes of **/scvelo-master/scvelo/plotting/velocity_graph** like following (Line 83)

> 1. change the zorder from 0 to 2 (let the zorder bigger than that of edges) first, 
> 2. and then add fontsize and figsize parameters statements (Set according to your needs):

Before changed:
`ax = scatter(adata, layer=layer, color=color, size=size, ax=ax, zorder=0, **kwargs)`
After changed:
`ax = scatter(adata, layer=layer, color=color, size=size, ax=ax, zorder=2, fontsize=10,figsize=(8,6), **kwargs)`

velocity graph and transition plot **before changes**:
![debug1](https://user-images.githubusercontent.com/82143598/216776836-0163d610-9f8e-4b6b-8eba-6d25f9fac2ff.png)
![debug2](https://user-images.githubusercontent.com/82143598/216776839-65a08734-5f90-4268-b802-4c965f87f853.png)

velocity graph and transition plot **after changes**:
![win！](https://user-images.githubusercontent.com/82143598/216776865-2b956e75-f4be-48c5-bc47-e6000aeaf331.png)
the following plot should add one more zorder statement (zorder=3)
`x, y = scv.utils.get_cell_transitions(adata, basis='umap', starting_cell=70)`
`ax = modified_velocity_graph(adata, arrows=True, c='lightgrey', edge_width=.05, show=False)`
`ax = scv.pl.scatter(adata, x=x, y=y, s=120, c='ascending', cmap='gnuplot', ax=ax, zorder=3)`
![拿下！](https://user-images.githubusercontent.com/82143598/216776867-67e6fb08-88a4-4c8d-8247-b80db4f97557.png)

**Although this solution could generate the correct figures, it only works in the situation of setting `arrows=True`, setting `arrows=False` will still lead to the wrong layer overlapping. And I would like to mention, setting `arrows=True` will cost more time (more computing power is cost in the process of computing directed graph?). I don't know how to correct it fundamentally, but at least now, we have the correct output figures. Hope it would be solved fundamentally asap. Thanks all**
